File: backend/app/schemas/debrief.py
"""Debrief (coaching report) schemas."""
from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class DimensionScores(BaseModel):
    clarity: int = Field(ge=0, le=100)
    deescalation: int = Field(ge=0, le=100)
    info_gathering: int = Field(ge=0, le=100)
    escalation: int = Field(ge=0, le=100)
    status_communication: int = Field(ge=0, le=100)


class LostControlMoment(BaseModel):
    quote: str
    issue: str
    better: str


class DebriefContent(BaseModel):
    strengths: list[str]
    lost_control: list[LostControlMoment]
    missed_information: list[str]
    escalation_assessment: str
    status_communication_assessment: str
    dimension_scores: DimensionScores


class DebriefOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    session_id: str
    summary: str
    overall_grade: str
    content: DebriefContent
    created_at: datetime

File: shared/types/debrief.ts
// TS mirror of the backend debrief schemas (app/schemas/debrief.py).

export interface DimensionScores {
  clarity: number;
  deescalation: number;
  info_gathering: number;
  escalation: number;
  status_communication: number;
}

export interface LostControlMoment {
  quote: string;
  issue: string;
  better: string;
}

export interface DebriefContent {
  strengths: string[];
  lost_control: LostControlMoment[];
  missed_information: string[];
  escalation_assessment: string;
  status_communication_assessment: string;
  dimension_scores: DimensionScores;
}

export interface DebriefOut {
  session_id: string;
  summary: string;
  overall_grade: string;
  content: DebriefContent;
  created_at: string;
}
